src/main/java/io/neonbee/data/DataVerticle.java

```diff
@@ -104,6 +299,18 @@
      */
     public abstract String getName();
 
+    /**
+     * Returns how this verticle handles the response metadata. If MANUAL_PROCESS is returned, it is the job of the
+     * verticle to handle the merging and determine the returned metadata. If AUTO_MERGE is returned, the metadata of
+     * all invoked verticles will be automatically merged and returned. The default is set to MANUAL_PROCESS, each
+     * verticle can overwrite this method to use a different policy.
+     *
+     * @return how this verticle handles the response metadata.
+     */
+    public ResponseDataPropagationPolicy getResponseDataPropagationPolicy() {
+        return ResponseDataPropagationPolicy.MANUAL_PROCESS;
+    }
+
```

Comment:
As we now have a context.propagateresponseData() method, do we really need this? Seem like overkill to me: So in order to do the propagation, I would have to:

```java
@Override
public ResponseDataPropagationPolicy getResponseDataPropagationPolicy() {
        return ResponseDataPropagationPolicy.AUTO_MERGE;
}
```

And this causes some "magic" to happen. Instead I could simply (in my already implemented `retrieveData` call:

```java
@Override
public Future<?> retrieveData() {
        // ...
        context.propagateResponseData();
        // ...
}
```

Before returning the method. Doesn't the second one fully cover the whole functionallity much more easily and more clear to the end user, because he can influence when and where he wants to do the propagation?

Reply:
Done